torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -856,9 +1016,62 @@ void SegmentCandidateFinder::findSegments() {
     mergeNodes();
   }
 
+  finalMerge();
+
   finalize();
 }
 
+void SegmentCandidateFinder::finalMerge() {
+  BackwardReachability reachability(segmented_fusion_.get());
+
+  bool merged_nodes = true;
+  while (merged_nodes) {
+    // One merge per iteration
+    for (auto g : groups()) {
+      // populate consumers
+      std::unordered_map<SegmentedGroup*, SegmentedEdge*> edge_map;
+      for (auto c : g->consumer_edges) {
+        edge_map.insert({c->to, c});
+      }
+      std::vector<SegmentedGroup*> all_consumers;
```

Comment:
`all_consumers_of_consumer` otherwise this also gets a bit confusing.

Reply:
This is actually `all_consumers_of_producer`. Renamed it accordingly and sorry for the confusion.